samza-sql/src/main/java/org/apache/samza/sql/interfaces/SqlSystemStreamConfig.java

```diff
@@ -38,20 +42,43 @@
 
   private final String samzaRelConverterName;
   private final SystemStream systemStream;
+
+  private final String source;
   private String relSchemaProviderName;
 
-  public SqlSystemStreamConfig(String systemName, String streamName, Config systemConfig) {
+  private Config config;
+
+  private List<String> sourceParts;
+
+  public SqlSystemStreamConfig(String systemName, String streamName, String source, Config systemConfig) {
+    this(systemName, streamName, source, Arrays.asList(systemName, streamName), systemConfig);
+  }
 
+  public SqlSystemStreamConfig(String systemName, String streamName, String source, List<String> sourceParts,
```

Comment:
I wanted to keep the logic of converting the source String into source parts in the source resolver plugin. So that plugins can parse however they want based on the business needs. But i do see the problem here that we are parsing the redundant information. Let me take in the sourceParts and create a canonical source String that can be used to lookup the RelSchemaProviders and converters.